cpp/tests/streams/merge_test.cpp

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/table_utilities.hpp>
+#include <cudf_test/testing_main.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <cudf/merge.hpp>
+#include <cudf/sorting.hpp>
+#include <cudf/table/table.hpp>
+#include <cudf/types.hpp>
+
+#include <vector>
+
+template <typename T>
+class MergeTest_ : public cudf::test::BaseFixture {};
+
+TYPED_TEST_SUITE(MergeTest_, cudf::test::FixedWidthTypes);
+
+TYPED_TEST(MergeTest_, MergeIsZeroWhenShouldNotBeZero)
+{
+  using columnFactoryT = cudf::test::fixed_width_column_wrapper<TypeParam, int32_t>;
+
+  columnFactoryT leftColWrap1({1, 2, 3, 4, 5});
+  cudf::test::fixed_width_column_wrapper<TypeParam> rightColWrap1{};
+
+  std::vector<cudf::size_type> key_cols{0};
+  std::vector<cudf::order> column_order;
+  column_order.push_back(cudf::order::ASCENDING);
+  std::vector<cudf::null_order> null_precedence(column_order.size(), cudf::null_order::AFTER);
+
+  cudf::table_view left_view{{leftColWrap1}};
+  cudf::table_view right_view{{rightColWrap1}};
+  cudf::table_view expected{{leftColWrap1}};
+
+  auto result = cudf::merge({left_view, right_view}, key_cols, column_order, null_precedence);
```

Comment:
Same change at other places as well.
```suggestion
  auto result = cudf::merge({left_view, right_view}, key_cols, column_order, null_precedence);
  auto result = cudf::merge({left_view, right_view}, key_cols, column_order, null_precedence, cudf::test::get_default_stream());
```